### DISCUSSION TITLE
How to customize specific controls from components-rsuite?

### DISCUSSION THREAD

**Discussion by shishir-korde-tech9** [optimajet/formengine#21] [Q&A]
I would like to understand how do I customize only specific controls from components-rsuite? I don't mean only styling, but behaviors too. eg., doing a custom action on say click of add button of container.
Use case here could be say, We show a popup on click of container that shows all the controls registered. However in my app I want that popup to show only unused controls, or I want to show a different popup hence a predefined custom action.

Another similar question here is, can I reuse the existing action provided by the library on my own controls? eg, I want the same behaviour of add button of container but I just want to add that behaviour on my own custom button?

**Comment by optimajet** [comment]
Hi @shishir-korde-tech9,

> I would like to understand how do I customize only specific controls from components-rsuite?

The components in the `components-rsuite` package are just a thin wrapper around the React Suite library. You can simply define your own components in place of the components you already have (all or just part of them). Just filter out the components you don't need in `rSuiteComponents` (https://formengine.io/documentation/api-reference/modules/react_form_builder_components_rsuite#rsuitecomponents) (for Designer) or `models` (https://formengine.io/documentation/api-reference/modules/react_form_builder_components_rsuite#models) and substitute your components with whatever logic, behaviour and styles you need.

> Another similar question here is, can I reuse the existing action provided by the library on my own controls? eg, I want the same behaviour of add button of container but I just want to add that behaviour on my own custom button?

Not quite sure what reuse difficulties you encountered?
